src/Domain/AccountPackage.php

```diff
@@ -0,0 +1,72 @@
+<?php declare(strict_types = 1);
+
+namespace SandwaveIo\BaseKit\Domain;
+
+final class AccountPackage implements DomainObjectInterface
+{
+    /**
+     * @param array<mixed> $startDateTime
+     * @param array<mixed> $endDateTime
+     * @param array<mixed> $update
+     */
+    public function __construct(
+        public int $ref,
+        public array $startDateTime,
+        public array $endDateTime,
+        public array $update,
+        public int $deleteOnExpiry,
+        public bool $isFree,
+        public int $billingPeriodMonths,
+        public bool $isActive,
+        public Package $package,
+        public ?int $templateGroupRef,
+        public ?int $displayOrder,
+    ) {
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function toArray(): array
+    {
+        return [
+            'ref' => $this->ref,
+            'startDateTime' => $this->startDateTime,
+            'endDateTime' => $this->endDateTime,
+            'update' => $this->update,
+            'deleteOnExpiry' => $this->deleteOnExpiry,
+            'isFree' => $this->isFree,
+            'billingPeriodMonths' => $this->billingPeriodMonths,
+            'isActive' => $this->isActive,
+            'package' => $this->package,
```

Comment:
Currently with the toArray call, I myself would expect that would be done recursively. If you do agree on this, then I would make the following change:
```suggestion
            'package' => $this->package->toArray(),
```

If you do not agree, then keeping it like it is, is no problem.

Reply:
I agree, as some of them can be null I made some null safe, resolving the one below as well as it is the same